src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcHistogram.scala

```diff
@@ -14,23 +14,30 @@
  *
  */
 
-
-
 package com.amazon.deequ.analyzers.jdbc
 
 import java.sql.ResultSet
 
 import com.amazon.deequ.analyzers.jdbc.Preconditions.{hasColumn, hasTable}
-import com.amazon.deequ.analyzers.runners.{EmptyStateException, MetricCalculationException}
+import com.amazon.deequ.analyzers.runners.{EmptyStateException, IllegalAnalyzerParameterException, MetricCalculationException}
 import com.amazon.deequ.metrics._
 
 import scala.util.{Failure, Try}
 
-case class JdbcHistogram(column: String)
+case class JdbcHistogram(column: String,
+                         binningFunc: Option[Any => Any] = None,
+                         maxDetailBins: Integer = JdbcHistogram.MaximumAllowedDetailBins)
   extends JdbcAnalyzer[JdbcFrequenciesAndNumRows, HistogramMetric] {
 
+  private[this] val PARAM_CHECK: Table => Unit = { _ =>
+    if (maxDetailBins > JdbcHistogram.MaximumAllowedDetailBins) {
+      throw new IllegalAnalyzerParameterException(s"Cannot return histogram values for more " +
+        s"than ${JdbcHistogram.MaximumAllowedDetailBins} values")
+    }
+  }
+
   override def preconditions: Seq[Table => Unit] = {
-    hasTable() :: hasColumn(column) :: Nil
+    PARAM_CHECK :: hasTable() :: hasColumn(column) :: Nil
```

Comment:
I'd suggest to stick to the precondition naming pattern here. All caps is usually used for constants, not for functions. Maybe name it "hasValidParams" or "hasValidMaxDetailParam"

Reply:
So, Amazon did this, too. They named it PARAM_CHECKS so just add an 's'